webapp/src/reducers.tsx

```diff
@@ -11,7 +11,7 @@ export const openBulkInviteChannelModal = (channelId: string) => {
     };
 };
 
-const bulkAddChannelModalVisible = (state = false, action) => {
+const bulkAddChannelModalVisible = (state = false, action: AnyAction) => {
```

Comment:
I was trying to figure out the "proper" way to declare a reducer without explicitly referencing the `AnyAction` type. There is an available [Reducer](https://github.com/reduxjs/redux/blob/8e58a32a78aba1e1bd8829e827a0c0b2c72d72e3/src/types/reducers.ts#L30) type to declare for the function type, used like this:

```ts
const bulkAddChannelModalVisible: Reducer<boolean> = (state = false, action) => {
```

That automatically infers the type of `action` to be `AnyAction`. When possible, I generally prefer this sort of type inference, as we don't really care about the `AnyAction` itself. We care about the rules of a `Reducer`.

---

This may be overkill, but you can also do this to declare the action type as well:

```ts
type BulkAddChannelModalAction = {
    type: typeof action_types['OPEN_BULK_ADD_CHANNEL_MODAL'] | typeof action_types['CLOSE_BULK_ADD_CHANNEL_MODAL'];
};

// or

type BulkAddChannelModalAction = {
    type: 'OPEN_BULK_ADD_CHANNEL_MODAL' | 'CLOSE_BULK_ADD_CHANNEL_MODAL';
};

const bulkAddChannelModalVisible: Reducer<boolean, BulkAddChannelModalAction> = (state = false, action) => {
```

and change this line to have `as const`

https://github.com/mattermost/mattermost-plugin-bulk-invite/blob/f1b32ae4021d9fb5981f3e36e722f17a7110fe85/webapp/src/action_types.ts#L6

```ts
}; as const
```

Reply:
I like this, switching to that to ease up code visiblity.